agency/README.md

```diff
@@ -41,8 +41,17 @@ Path Params:
 | ------------ | ---- | ----------------- | ------------------------------------------- |
 | `device_id` | UUID  | Optional          | If provided, retrieve the specified vehicle |
 
+Query Params:
+
+| Param        | Type | Required/Optional | Description                                 |
+| ------------ | ---- | ----------------- | ------------------------------------------- |
+| `skip` | Integer  | Optional          | If provided, offset into list |
+| `take` | Integer  | Optional          | If provided, number of vehicles to load |
```

Comment:
With this change, we'll need to add support for pagination to the spec. However, this proposal feels like an implementation detail - in the provider API, [compliance with JSON API spec for pagination is required](https://github.com/CityOfLosAngeles/mobility-data-specification/tree/0.3.0/provider#pagination), and that's been sufficient to get everyone on the same page. Might we consider taking a similar approach here?

Reply:
Bleah, I am not a fan that pagination spec.  It's too prescriptive and heavy-weight.

Reply:
A limit/offset pagination API is probably not ideal since changes to the underlying data while the client is traversing the list can easily cause omitted or repeated records. I sympathize with not loving JSON API, but we'd need something more robust than this anyway, and I don't know how much sense it makes for MDS to include more than one pagination system.

Reply:
All pagination systems are vulnerable to underlying data changes, not just limit/offset.  JSON API doesn't address that:

> Note: JSON:API is agnostic about the pagination strategy used by a server.

It's pretty simple to pick a sort criteria amenable to consistent results (e.g. creation-date of the record).

Reply:
Point taken on multiple pagination strategies.  I just despise JSON API's.

Reply:
Vehicle records are big enough that a single query for, say, 20K vehicles is out of the question, so IMHO pagination is a firm requirement.  I guess I'll suck it up and use the established one.

Reply:
Using cursor-based pagination makes it possible to avoid these problems, although still not trivial. But it sounds like you're convinced, so I'll skip that argument 😅

Reply:
I'm still going to use skip/take.  I'm just going to make up a page size in the reference implementation and construct my link URLs accordingly.  We can put off further bikeshedding until another day.

Reply:
(The bikeshedding crack is squarely directed at myself.)

Reply:
Sure, yeah, it's fine for the reference implementation to use skip/take as long as the spec allows implementors to use something fancier :+1:

Reply:
Smart

Reply:
Sandbox updated with new behavior